src/components/Form/Form.jsx

```diff
@@ -15,12 +16,37 @@ export const Form = () => {
     lastDate: null,
   });
 
+  const [value] = useCollection(getAll());
+
+  const isItemDuplicated = (name) => {
+    const regex = /[!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~]/g;
+
+    if (value) {
+      const item = value.docs.filter(
+        (doc) =>
+          doc.data().name.toLowerCase().replace(regex, '') ===
+          name.toLowerCase().replace(regex, ''),
+      );
+      return item.length ? true : false;
+    } else {
+      return false;
+    }
+  };
+
   const sendToFB = (e) => {
     e.preventDefault();
-    const itemLength = values?.nameItem;
+    const itemName = values?.nameItem;
```

Comment:
what do you think to use destructuring here like this:
`const { nameItem,  selectTime } = values`

Reply:
I believe that would let them use the optional chaining. It's unlikely that values would become undefined, but I do like the protection of the optional chaining. However, I also feel like I'm optional chaining like absolutely everything these days and I'm not sure if there's a tradeoff that I'm missing somewhere...